### PR TITLE
Edits to error-handling chapter

### PR DESCRIPTION
**Pull request by gvwilson**
Depends on #398.

### CODE REVIEW

File: py-rse/errors.Rmd [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 People will give our programs options that aren't supported
 or ask those programs to read files that don't exist.
 Our code will also inevitably contain bugs,
 so we should plan from the start to catch and handle errors.
-
-Your Zipf's Law project should now include:
+In this chapter,
+we will explore how errors are represented in programs
+and what we should do with them.
+The Zipf's Law project should now include:
```
Comment (DamienIrving) [inline py-rse/errors.Rmd:11]: In the summary of the project tree `mymodule.py` needs to be updated to `utilities.py` and `rcparams.yml` to `plotparams.yml`.

(We've failed to update `mymodule.py` in previous chapters too.)

File: py-rse/errors.Rmd [hunk 2]
```diff
@@ -27,19 +30,14 @@ Your Zipf's Law project should now include:
...
 Most modern programming languages use [exceptions][exception] for error handling.
 As the name suggests,
 an exception is a way to represent an exceptional or unusual occurrence
 that doesn't fit neatly into the program's expected operation.
-The code below shows a very simple example that tests different denominators and
-reports attempts to divide by zero:
+The code below shows a very simple example
+that uses exceptions to report attempts to divide by zero:
 
 ```python
```
Comment (DamienIrving) [inline py-rse/errors.Rmd:42]: Following #342, this code block needs to be updated to use f-strings.

File: py-rse/errors.Rmd [hunk 3]
```diff
@@ -187,20 +191,17 @@ except ArithmeticError as error:
...
-e.g. if you try to read a file that does not exist,
-the error handling inside the `open` function throws a `FileNotFoundError`:
+We can now go ahead and add error handling to our Zipf's Law code.
+Some is already built in:
+for example,
+if we try to read a file that does not exist,
+the `open` function throws a `FileNotFoundError`:
 
 ```shell
 python bin/collate.py results/none.csv
```
Comment (DamienIrving) [inline py-rse/errors.Rmd:207]:
```suggestion
python bin/collate.py results/none.csv results/dracula.csv
```
Might be a more authentic use of `collate.py` if more than one input file is listed?

File: py-rse/errors.Rmd [hunk 4]
```diff
@@ -597,17 +568,17 @@ logging.info('Processing files...')
...
 
-We can also configure logging to send messages to a file instead of to standard error
+We can also configure logging to send messages to a file instead of standard error
 using `logging.basicConfig`.
-This has to be done before we make any logging calls,
-it's not retroactive.
-We can use the same function to set the logging level:
+(This has to be done before we make any logging calls:
+it's not retroactive.)
+We can aso use that function to set the logging level;
```
Comment (DamienIrving) [inline py-rse/errors.Rmd:581]:
```suggestion
We can also use that function to set the logging level;
```

File: py-rse/errors.Rmd [hunk 5]
```diff
@@ -597,17 +568,17 @@ logging.info('Processing files...')
 for file_name in args.infiles:
     logging.debug('Reading in {file_name}...')
```
Comment (DamienIrving) [inline py-rse/errors.Rmd:569]:
```suggestion
    logging.debug(f'Reading in {file_name}...')
```

File: py-rse/errors.Rmd (hunk 1 above)
Comment (gvwilson) [reply py-rse/errors.Rmd:11]: Good catch - amended in all files.

File: py-rse/errors.Rmd (hunk 2 above)
Comment (gvwilson) [reply py-rse/errors.Rmd:42]: -  Fixed those that can be converted (the error messages looked up in the dictionary can't be f-strings).
-  Altered `style.Rmd` to match.

File: py-rse/errors.Rmd (hunk 3 above)
Comment (gvwilson) [reply py-rse/errors.Rmd:207]: :+1:

File: py-rse/errors.Rmd (hunk 4 above)
Comment (gvwilson) [reply py-rse/errors.Rmd:581]: :+1:

File: py-rse/errors.Rmd (hunk 5 above)
Comment (gvwilson) [reply py-rse/errors.Rmd:568]: :+1:

**Comment by gvwilson** [comment]
Discussion of update to `collate.py` moved to #412.

**Pull request merged**